src/microbe_stage/systems/FluidCurrentsSystem.cs

```diff
@@ -116,7 +179,18 @@ protected override void Update(float delta, in Entity entity)
         var pos = new Vector2(position.Position.X, position.Position.Z);
         var vel = VelocityAt(pos) * Constants.MAX_FORCE_APPLIED_BY_CURRENTS;
 
-        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta;
+        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta
+            * entity.Get<CurrentAffected>().EffectStrength;
         physicsControl.PhysicsApplied = false;
     }
+
+    private float GetPixel(float x, float y, float z, Image[] array)
```

Comment:
This should be marked as aggressive inlining as I think that might help a bit with the performance.